Expensify/App
# Move deploy specifics to github actions for desktop builds

### Details
Changed `npm run desktop-build` to not deploy and instead just build a production version of the app without signing, etc.

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes https://github.com/Expensify/Expensify/issues/144258

### Tests
1. Run `npm run desktop-build` Verify no errors 👍 
3.  Run `npm run desktop-build -- --publish always -c.afterSign=desktop/notarize.js` Verify you get an error `No authentication properties provided (e.g. appleId, appleApiKey)` (GitHub actions will have this set, so it will deploy!)

### Tested On
